File: frontEnd/src/types/dto/stepDTO/ExtractStepDTO.ts
// 提取操作
import {ExtractType} from "@/types/enum/extract/ExtractType";
import {ElementExtractValueType} from "@/types/enum/extract/ElementExtractValueType";
import {PageExtractValueType} from "@/types/enum/extract/PageExtractValueType";
import {StepBaseDTO} from "@/types/dto/base/StepBaseDTO";
import {ElementDTO} from "@/types/dto/ElementDTO";

// 提取操作
export class ExtractStepDTO extends StepBaseDTO {
    stepType: string = 'EXTRACT'; // 步骤类型
    variableName: string = ''; // 变量名
    extractType: ExtractType | string = "PAGE"; // 提取类型
    element?: ElementDTO = new ElementDTO(); // 提取的目标元素
    elementExtractType?: ElementExtractValueType | string = "TEXT"; // 元素提取类型
    elementAttribute?: string = "value";
    pageExtractType?: PageExtractValueType | string = "TITLE"; // 页面提取类型
}

File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/step/ExtractStepDTO.java
package com.mokatest.platform.demos.domain.ui.dto.step;// 提取操作


import com.mokatest.platform.demos.domain.ui.uiEnum.extract.ElementExtractValueType;
import com.mokatest.platform.demos.domain.ui.uiEnum.extract.ExtractType;
import com.mokatest.platform.demos.domain.ui.uiEnum.extract.PageExtractValueType;
import com.mokatest.platform.demos.domain.ui.dto.ElementDTO;
import lombok.Data;

// 提取操作
@Data
public class ExtractStepDTO extends StepBaseDTO {
    private String stepType; // 步骤类型
    private String variableName; // 变量名
    private ExtractType extractType; // 提取类型
    private ElementDTO element; // 提取的目标元素
    private ElementExtractValueType elementExtractType; // 元素提取类型
    private String elementAttribute;
    private PageExtractValueType pageExtractType; // 页面提取类型

    private String cookieName; // cookie名称
}

